File: services/lucy-dotnet/src/Lucy.AuthPayment.Api/Modules/Payment/Models/ContentPurchase.cs
namespace lucy_auth_payment.Modules.Payment.Models;

public class ContentPurchase
{
    public string PurchaseId { get; set; } = null!;
    public string ContentId { get; set; } = null!;
    public string BuyerUserId { get; set; } = null!;
    public string SellerUserId { get; set; } = null!;
    public decimal PriceAmount { get; set; }
    public string? BuyerWalletTransactionId { get; set; }
    public string? SellerWalletTransactionId { get; set; }
    public DateTime PurchasedAt { get; set; } = DateTime.Now;
}

File: lucy-web-stack/lucy-backend/src/main/java/com/lucy/backend/content/creator/entity/ContentPurchase.java
package com.lucy.backend.content.creator.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "ContentPurchases")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ContentPurchase {

    @Id
    @Column(name = "PurchaseId", length = 50)
    private String purchaseId;

    @Column(name = "ContentId", length = 50, nullable = false)
    private String contentId;

    @Column(name = "BuyerUserId", length = 50, nullable = false)
    private String buyerUserId;

    @Column(name = "SellerUserId", length = 50, nullable = false)
    private String sellerUserId;

    @Column(name = "PriceAmount", nullable = false)
    private BigDecimal priceAmount;

    @Column(name = "PurchasedAt", nullable = false)
    private LocalDateTime purchasedAt;
}
